### DISCUSSION TITLE
[Cinnamon] [Consider a New Approach Request] change some 3rd party (Gnome tools for Mate tools).

### DISCUSSION THREAD

**Discussion by fcolecumberri** [linuxmint/discussions#399] [Ideas]
For the last not-so-few years Gnome has been taking decisions in terms of user experience that are non-backward compatible (and questionable, depending on who you ask). This should not be a problem for Cinnamon since these are 2 different projects. However Cinnamon by default uses some Gnome tools, like `File-roller`, `Gnome-calculator`, etc. Making Cinnamon user experience prone to issues each time Gnomes makes a non-backward compatible change.

Before writing this I searched on https://github.com/linuxmint/cinnamon/issues to see if this is only my experience or if there are more people with this issue. By reading https://github.com/linuxmint/cinnamon/issues/11010 , https://github.com/linuxmint/cinnamon/issues/5367 , https://github.com/linuxmint/cinnamon/issues/3002 , etc. I can confirm this is not an isolated case (However I don't know how much of an issue this is).

I am sure some of this catch-ups with Gnome should have brought some headache to more than one developer.

In the case of my system I have been able to fix this by installing Mate's `Engrampa`, `Mate-calc`, etc. and using those instead of the defaults.

Mate tools IMHO interacts better with Cinnamon, not just with theming, but for some reason drag-n-drop stop working with `File-roller`/`Nemo` (while `Engrampa`/`Nemo` works perfectly). Actually I was going to write an issue about this before trying `Engrampa`.

In conclusion, I think switching the default 3rd party tools could be beneficial for the project so I would like this to be at least considered.

**Comment by mtwebster** [comment]
The latest Mint blog (today) discusses a bit of this:
https://blog.linuxmint.com/?p=4675

**Reply by fcolecumberri** [reply]
Nice, but, for example, the blog is talking about forking `GNOME Calculator`, `Baobab`, `System Monitor` and `File Roller`. Why forking those instead of forking Mate's software? Mate has already forks of those 4.

Mate's software's main issue according to that blog post is that it has written MATE all over the place, which I guess is something not that difficult to change compared with Gnome stuff (Also I don't understand why is that a deal breaker).

Not just that, but Mate seems to have focused more on functionality than Gnome, as an example, https://gitlab.gnome.org/GNOME/file-roller/-/issues/179 it seems like `File Roller` is importing functionality from `Engrampa` (Not the other way around). Another example is `Xed` which I guess you forked from `Pluma` instead of `Gedit` for a good reason.

Thanks for sharing that blog post.
